app/mailers/ddr/batch/batch_processor_run_mailer.rb

```diff
@@ -1,19 +1,22 @@
 module Ddr::Batch
-
   class BatchProcessorRunMailer < ActionMailer::Base
 
-    default :from => "noreply@duke.edu"
-
     def send_notification(batch)
       @batch = batch
       @title = "Batch Processor Run #{@batch.status}"
       @host = `uname -n`.strip
       @subject = "[#{@host}] #{@title}"
-      from = "#{`echo $USER`.strip}@#{@host}"
       attachments["details.txt"] = File.read(@batch.logfile.path)
-      mail(from: from, to: @batch.user.email, subject: @subject)
+      mail(from: from_address, to: @batch.user.email, subject: @subject)
     end
 
-  end
+    private
 
-end
\ No newline at end of file
+    def from_address
+      Rails.application.config.action_mailer.default_options[:from]
+    rescue
```

Comment:
It would be more informative to raise an exception explaining that one is expected to set that option, if that's in fact the case.  Otherwise you have to track back from why is the from address nil.